### PR TITLE
feat(base): add util for chunked parallelize loop

### PR DESCRIPTION
**Pull request by dongchangYoo**
It provides a `Chunked Parallelize Loop` feature using OpenMP.

### CODE REVIEW

**Review by chokobole — APPROVED** [review]
LGTM

File: tachyon/math/polynomials/univariate/radix2_evaluation_domain.h [hunk 1]
```diff
@@ -145,17 +146,13 @@ class Radix2EvaluationDomain : public UnivariateEvaluationDomain<F, MaxDegree> {
...
-                    [](absl::Span<F> chunk) { return chunk; });
-      OPENMP_PARALLEL_FOR(size_t i = 0; i < chunks_vector.size(); ++i) {
-        absl::Span<F> chunks = chunks_vector[i];
-        const F& v = chunks[0];
-        for (size_t j = 1; j < chunks.size(); ++j) {
-          chunks[j] = v;
-        }
-      }
+      base::ParallelizeByChunkSize(evals.evaluations_, duplicity_of_initials,
+                                   [](absl::Span<F> chunk, size_t i) {
```
Comment (chokobole) [inline tachyon/math/polynomials/univariate/radix2_evaluation_domain.h:150]:
```suggestion
                                   [](absl::Span<F> chunk) {
```

**Review by Insun35 — APPROVED** [review]
LGTM

File: tachyon/base/parallelize_unittest.cc [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+          v += 1;
+        }
+      });
+  EXPECT_EQ(expected, test_in);
+  EXPECT_THAT(chunk_indices, testing::UnorderedElementsAreArray({0, 1, 2}));
+
+  expected = {3, 4, 5, 6, 7, 8};
+  chunk_indices = base::CreateVector(test_in.size() / chunk_size, size_t{0});
+  std::vector<size_t> chunk_sizes =
+      base::CreateVector(test_in.size() / chunk_size, size_t{0});
```
Comment (TomTaehoonKim) [inline tachyon/base/parallelize_unittest.cc:42]:
```suggestion
      base::CreateVector(test_in.size() / chunk_size, 0);
```

File: tachyon/base/parallelize_unittest.cc [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        chunk_indices[chunk_offset] = chunk_offset;
+        for (int& v : chunk) {
+          v += 1;
+        }
+      });
+  EXPECT_EQ(expected, test_in);
+  EXPECT_THAT(chunk_indices, testing::UnorderedElementsAreArray({0, 1, 2}));
+
+  expected = {3, 4, 5, 6, 7, 8};
+  chunk_indices = base::CreateVector(test_in.size() / chunk_size, size_t{0});
```
Comment (TomTaehoonKim) [inline tachyon/base/parallelize_unittest.cc:40]:
```suggestion
  chunk_indices = base::CreateVector(test_in.size() / chunk_size, 0);
```

File: tachyon/base/parallelize_unittest.cc [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+  std::vector<size_t> chunk_sizes =
+      base::CreateVector(test_in.size() / chunk_size, size_t{0});
+  ParallelizeByChunkSize(
+      test_in, chunk_size,
+      [chunk_size, &chunk_indices, &chunk_sizes](
+          absl::Span<int> chunk, size_t chunk_offset, size_t chunk_size_in) {
```
Comment (TomTaehoonKim) [inline tachyon/base/parallelize_unittest.cc:46]:
```suggestion
          absl::Span<int> chunk, size_t chunk_index, size_t chunk_size_in) {
```

File: tachyon/base/parallelize_unittest.cc [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+                           EXPECT_EQ(chunk.size(), chunk_size);
+                           for (int& v : chunk) {
+                             v += 1;
+                           }
+                         });
+  EXPECT_EQ(expected, test_in);
+
+  expected = {2, 3, 4, 5, 6, 7};
+  std::vector<size_t> chunk_indices =
+      base::CreateVector(test_in.size() / chunk_size, size_t{0});
```
[the remark this answers is not in the document]
Comment (dongchangYoo) [reply tachyon/base/parallelize_unittest.cc:26]: If we don't type cast to size_t, a build error occurs, so we will keep it as is.

File: tachyon/base/parallelize_unittest.cc (hunk 2 above)
Comment (dongchangYoo) [reply tachyon/base/parallelize_unittest.cc:42]: If we don't type cast to size_t, a build error occurs, so we will keep it as is.

File: tachyon/base/parallelize_unittest.cc (hunk 3 above)
Comment (dongchangYoo) [reply tachyon/base/parallelize_unittest.cc:40]: If we don't type cast to size_t, a build error occurs, so we will keep it as is.

**Review by TomTaehoonKim — APPROVED** [review]
LGTM

**Pull request merged**